composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/BookingsView.kt

```diff
@@ -0,0 +1,174 @@
+package com.overdrive.cruiser.views
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.overdrive.cruiser.models.Booking
+import com.overdrive.cruiser.models.BookingsViewModel
+import com.overdrive.cruiser.models.Spot
+import cruiser.composeapp.generated.resources.Res
+import cruiser.composeapp.generated.resources.booking_image
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+import org.jetbrains.compose.resources.painterResource
+
+
+@Composable
+fun BookingsView(savedSpotsViewModel: BookingsViewModel) {
+    val allBookings by savedSpotsViewModel.bookings.collectAsState()
+    val currentTime = remember { Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault()) }
+
+    val upcomingBookings = allBookings.filter { it.endTime > currentTime }
+    val pastBookings = allBookings.filter { it.endTime < currentTime }
+
+    LaunchedEffect(Unit) {
+        savedSpotsViewModel.updateBookings()
+    }
+
+    Column {
```

Comment:
probably want to add Modifier.background(color = Color(0xFFF5F5F5)) to the column so it matches the other pages

Reply:
Good catch, updated this. Eventually this should be shared somewhere but I don't know where yet.